src/protocol/vm/utils.rs

```diff
@@ -260,22 +267,24 @@ pub async fn get_code_for_contract(
 
     // Parse the address
     let addr: H160 = address.parse().map_err(|_| {
-        SimulationError::from(RpcError::InvalidRequest(format!(
-            "Failed to parse address: {}",
-            address
-        )))
+        SimulationError::RetryDifferentInput(format!("Failed to parse address: {}", address))
```

Comment:
Unsure about this one tbh.. I would limit the try different input things to calldata - not necessary any input to any function involved in here. Tbh if a bad address is passed somewhere it is much more likely to be a bug than a temporary failure.